acp/source_hub_client.go

```diff
@@ -336,45 +337,52 @@ func (a *sourceHubBridge) IsDocRegistered(
 
 func (a *sourceHubBridge) CheckDocAccess(
 	ctx context.Context,
-	permission DPIPermission,
+	permissions []DPIPermission,
 	actorID string,
 	policyID string,
 	resourceName string,
 	docID string,
 ) (bool, error) {
-	isValid, err := a.client.VerifyAccessRequest(
-		ctx,
-		permission,
-		actorID,
-		policyID,
-		resourceName,
-		docID,
-	)
-	if err != nil {
-		return false, NewErrFailedToVerifyDocAccessWithACP(err, "Local", policyID, actorID, resourceName, docID)
-	}
+	var hasAccess bool = false
+	var err error
 
-	if isValid {
-		log.InfoContext(
+	for _, permission := range permissions {
+		hasAccess, err = a.client.VerifyAccessRequest(
 			ctx,
-			"Document accessible",
-			corelog.Any("PolicyID", policyID),
-			corelog.Any("ActorID", actorID),
-			corelog.Any("Resource", resourceName),
-			corelog.Any("DocID", docID),
-		)
-		return true, nil
-	} else {
-		log.InfoContext(
-			ctx,
-			"Document inaccessible",
-			corelog.Any("PolicyID", policyID),
-			corelog.Any("ActorID", actorID),
-			corelog.Any("Resource", resourceName),
-			corelog.Any("DocID", docID),
+			permission,
+			actorID,
+			policyID,
+			resourceName,
+			docID,
 		)
-		return false, nil
+
+		if err != nil {
+			return false, NewErrFailedToVerifyDocAccessWithACP(
+				err,
+				"Local",
+				permission.String(),
+				policyID,
+				actorID,
+				resourceName,
+				docID,
+			)
+		}
+
+		if hasAccess {
+			break
```

Comment:
todo: commented on this in `acp/acp.go`, will repeat:
I don't think it's very useful to return `true` if any permission is present.
There should be a good reason for this I believe.

Reply:
- [x] done